sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/avro/AvroIOIT.java

```diff
@@ -125,7 +144,40 @@ public void writeThenReadAll() {
         ParDo.of(new DeleteFileFn())
             .withSideInputs(consolidatedHashcode.apply(View.asSingleton())));
 
-    pipeline.run().waitUntilFinish();
+    PipelineResult result = pipeline.run();
+    result.waitUntilFinish();
+    gatherAndPublishMetricResults(result);
+  }
+
+  private void gatherAndPublishMetricResults(PipelineResult result) {
```

Comment:
Do you think we could reduce the duplication here? Given we have similar code in all tests, maybe we can have it in one place (namely `gatherAndPublishMetricResults` and `extractMetrics`.